app/services/notifications/notifiable_action/send.rb

```diff
@@ -34,7 +34,15 @@ def call
             notified_at: Time.current,
           )
         end
-        Notification.import! notifications
+        conflict_target = %i[notifiable_id notifiable_type user_id]
+        conflict_target << :action if action.present?
+        index_predicate = "action IS#{action.present? ? ' NOT ' : ' '}NULL"
```

Comment:
Tip: you can also use Arel to avoid writing this type of SQL manually:

```ruby
[32] pry(main)> action_field = Notification.arel_table[:action]
[33] pry(main)> puts action_field.eq(nil).to_sql
"notifications"."action" IS NULL
[34] pry(main)> puts action_field.not_eq(nil).to_sql
"notifications"."action" IS NOT NULL
```

I used it in the past as SQL builder, it's handy :)